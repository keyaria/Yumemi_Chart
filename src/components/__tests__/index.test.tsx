/* eslint-disable */
//@ts-nocheck
import * as React from "react"
import * as renderer from "react-test-renderer"
import { mocked } from 'ts-jest/utils';
import { Line, LineProps } from "@nivo/line"

import { render, fireEvent, waitFor } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import { StaticQuery } from "gatsby"
import Index, {IndexPage} from "../../pages/index"

import Graph from "../graph";
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({result: { boundaryYear: 2015 , data: [{label: "総人口", data: [{year: 1960, value: 599135},{year: 1965, value: 579853}, {year: 1970, value: 568777}]}] }}),
  })
);

beforeEach(() => {
  fetch.mockClear() as unknown as jest.Mock;
});
describe("Index", () => {

  it("renders the checkboxes", () => {

    const data = {
      prefecturesField: {
        prefData: "[{\"prefCode\":1,\"prefName\":\"北海道\"},{\"prefCode\":2,\"prefName\":\"青森県\"},{\"prefCode\":3,\"prefName\":\"岩手県\"},{\"prefCode\":4,\"prefName\":\"宮城県\"},{\"prefCode\":5,\"prefName\":\"秋田県\"},{\"prefCode\":6,\"prefName\":\"山形県\"},{\"prefCode\":7,\"prefName\":\"福島県\"},{\"prefCode\":8,\"prefName\":\"茨城県\"},{\"prefCode\":9,\"prefName\":\"栃木県\"},{\"prefCode\":10,\"prefName\":\"群馬県\"},{\"prefCode\":11,\"prefName\":\"埼玉県\"},{\"prefCode\":12,\"prefName\":\"千葉県\"},{\"prefCode\":13,\"prefName\":\"東京都\"},{\"prefCode\":14,\"prefName\":\"神奈川県\"},{\"prefCode\":15,\"prefName\":\"新潟県\"},{\"prefCode\":16,\"prefName\":\"富山県\"},{\"prefCode\":17,\"prefName\":\"石川県\"},{\"prefCode\":18,\"prefName\":\"福井県\"},{\"prefCode\":19,\"prefName\":\"山梨県\"},{\"prefCode\":20,\"prefName\":\"長野県\"},{\"prefCode\":21,\"prefName\":\"岐阜県\"},{\"prefCode\":22,\"prefName\":\"静岡県\"},{\"prefCode\":23,\"prefName\":\"愛知県\"},{\"prefCode\":24,\"prefName\":\"三重県\"},{\"prefCode\":25,\"prefName\":\"滋賀県\"},{\"prefCode\":26,\"prefName\":\"京都府\"},{\"prefCode\":27,\"prefName\":\"大阪府\"},{\"prefCode\":28,\"prefName\":\"兵庫県\"},{\"prefCode\":29,\"prefName\":\"奈良県\"},{\"prefCode\":30,\"prefName\":\"和歌山県\"},{\"prefCode\":31,\"prefName\":\"鳥取県\"},{\"prefCode\":32,\"prefName\":\"島根県\"},{\"prefCode\":33,\"prefName\":\"岡山県\"},{\"prefCode\":34,\"prefName\":\"広島県\"},{\"prefCode\":35,\"prefName\":\"山口県\"},{\"prefCode\":36,\"prefName\":\"徳島県\"},{\"prefCode\":37,\"prefName\":\"香川県\"},{\"prefCode\":38,\"prefName\":\"愛媛県\"},{\"prefCode\":39,\"prefName\":\"高知県\"},{\"prefCode\":40,\"prefName\":\"福岡県\"},{\"prefCode\":41,\"prefName\":\"佐賀県\"},{\"prefCode\":42,\"prefName\":\"長崎県\"},{\"prefCode\":43,\"prefName\":\"熊本県\"},{\"prefCode\":44,\"prefName\":\"大分県\"},{\"prefCode\":45,\"prefName\":\"宮崎県\"},{\"prefCode\":46,\"prefName\":\"鹿児島県\"},{\"prefCode\":47,\"prefName\":\"沖縄県\"}]"
      }
}

     const { getAllByTestId } = render(<Index data={data}/>)
   const nodes =  getAllByTestId("gatsby-logo");
    expect(nodes).toHaveLength(47);
});
it.skip("updates graph based on checkbox", async () => {
  const data = {
    prefecturesField: {
      prefData: "[{\"prefCode\":1,\"prefName\":\"北海道\"},{\"prefCode\":2,\"prefName\":\"青森県\"},{\"prefCode\":3,\"prefName\":\"岩手県\"},{\"prefCode\":4,\"prefName\":\"宮城県\"},{\"prefCode\":5,\"prefName\":\"秋田県\"},{\"prefCode\":6,\"prefName\":\"山形県\"},{\"prefCode\":7,\"prefName\":\"福島県\"},{\"prefCode\":8,\"prefName\":\"茨城県\"},{\"prefCode\":9,\"prefName\":\"栃木県\"},{\"prefCode\":10,\"prefName\":\"群馬県\"},{\"prefCode\":11,\"prefName\":\"埼玉県\"},{\"prefCode\":12,\"prefName\":\"千葉県\"},{\"prefCode\":13,\"prefName\":\"東京都\"},{\"prefCode\":14,\"prefName\":\"神奈川県\"},{\"prefCode\":15,\"prefName\":\"新潟県\"},{\"prefCode\":16,\"prefName\":\"富山県\"},{\"prefCode\":17,\"prefName\":\"石川県\"},{\"prefCode\":18,\"prefName\":\"福井県\"},{\"prefCode\":19,\"prefName\":\"山梨県\"},{\"prefCode\":20,\"prefName\":\"長野県\"},{\"prefCode\":21,\"prefName\":\"岐阜県\"},{\"prefCode\":22,\"prefName\":\"静岡県\"},{\"prefCode\":23,\"prefName\":\"愛知県\"},{\"prefCode\":24,\"prefName\":\"三重県\"},{\"prefCode\":25,\"prefName\":\"滋賀県\"},{\"prefCode\":26,\"prefName\":\"京都府\"},{\"prefCode\":27,\"prefName\":\"大阪府\"},{\"prefCode\":28,\"prefName\":\"兵庫県\"},{\"prefCode\":29,\"prefName\":\"奈良県\"},{\"prefCode\":30,\"prefName\":\"和歌山県\"},{\"prefCode\":31,\"prefName\":\"鳥取県\"},{\"prefCode\":32,\"prefName\":\"島根県\"},{\"prefCode\":33,\"prefName\":\"岡山県\"},{\"prefCode\":34,\"prefName\":\"広島県\"},{\"prefCode\":35,\"prefName\":\"山口県\"},{\"prefCode\":36,\"prefName\":\"徳島県\"},{\"prefCode\":37,\"prefName\":\"香川県\"},{\"prefCode\":38,\"prefName\":\"愛媛県\"},{\"prefCode\":39,\"prefName\":\"高知県\"},{\"prefCode\":40,\"prefName\":\"福岡県\"},{\"prefCode\":41,\"prefName\":\"佐賀県\"},{\"prefCode\":42,\"prefName\":\"長崎県\"},{\"prefCode\":43,\"prefName\":\"熊本県\"},{\"prefCode\":44,\"prefName\":\"大分県\"},{\"prefCode\":45,\"prefName\":\"宮崎県\"},{\"prefCode\":46,\"prefName\":\"鹿児島県\"},{\"prefCode\":47,\"prefName\":\"沖縄県\"}]"
    }
}
const { getAllByTestId, container } = render(<Index data={data}/>)
const nodes =  getAllByTestId("gatsby-logo");

fireEvent.click(nodes[0])

  await waitFor(() => {
    expect(container).toMatchSnapshot()
});
expect(container).toMatchSnapshot()



  const { component, queryByTestId } = render(<div style={{height: `500px`}} height="500px"><Graph data={data}/> </div>)
    expect(component).toBeInTheDocument()

});
it('should render a line chart', async () => {
    const data = [
        {
            id: '北海道',
            data: [
                {x: 1960, y: 1670454},
                {x: 1965, y: 1645135},
                {x: 1970, y: 1707026},
                {x: 1975, y: 1814305},
                {x: 1980, y: 1954305},
            ],
        },
    ]
    const { component,findByText, getAllByText, findAllByDisplayValue, queryAllByRole  } = render(<Graph width={500} height={300} data={data}  enableSlices="x" animate={false} />)

    expect(component).toMatchSnapshot();
    await waitFor(() => expect(findAllByDisplayValue('1960')).toBeTruthy());
});



});
