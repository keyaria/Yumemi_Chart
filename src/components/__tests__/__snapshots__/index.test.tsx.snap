// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`Index should render a basic line chart 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: relative;"
      >
        <svg
          height="300"
          role="img"
          width="500"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="transparent"
            height="300"
            width="500"
          />
          <g
            transform="translate(0,0)"
          >
            <g>
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="0"
                y1="0"
                y2="300"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="125"
                x2="125"
                y1="0"
                y2="300"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="250"
                x2="250"
                y1="0"
                y2="300"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="375"
                x2="375"
                y1="0"
                y2="300"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="500"
                x2="500"
                y1="0"
                y2="300"
              />
            </g>
            <g>
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="300"
                y2="300"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="269"
                y2="269"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="239"
                y2="239"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="208"
                y2="208"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="177"
                y2="177"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="146"
                y2="146"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="116"
                y2="116"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="85"
                y2="85"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="54"
                y2="54"
              />
              <line
                opacity="1"
                stroke="#dddddd"
                stroke-width="1"
                x1="0"
                x2="500"
                y1="24"
                y2="24"
              />
            </g>
            <g
              transform="translate(0,300)"
            >
              <g
                style="opacity: 1;"
                transform="translate(0,0)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="5"
                />
                <text
                  dominant-baseline="text-before-edge"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="middle"
                  transform="translate(0,10) rotate(0)"
                >
                  1960
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(125,0)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="5"
                />
                <text
                  dominant-baseline="text-before-edge"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="middle"
                  transform="translate(0,10) rotate(0)"
                >
                  1965
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(250,0)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="5"
                />
                <text
                  dominant-baseline="text-before-edge"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="middle"
                  transform="translate(0,10) rotate(0)"
                >
                  1970
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(375,0)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="5"
                />
                <text
                  dominant-baseline="text-before-edge"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="middle"
                  transform="translate(0,10) rotate(0)"
                >
                  1975
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(500,0)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="5"
                />
                <text
                  dominant-baseline="text-before-edge"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="middle"
                  transform="translate(0,10) rotate(0)"
                >
                  1980
                </text>
              </g>
              <line
                style="stroke: transparent; stroke-width: 1;"
                x1="0"
                x2="500"
                y1="0"
                y2="0"
              />
            </g>
            <g
              transform="translate(0,0)"
            >
              <g
                style="opacity: 1;"
                transform="translate(0,300)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  0
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,269)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  200000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,239)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  400000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,208)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  600000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,177)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  800000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,146)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  1000000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,116)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  1200000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,85)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  1400000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,54)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  1600000
                </text>
              </g>
              <g
                style="opacity: 1;"
                transform="translate(0,24)"
              >
                <line
                  style="stroke: #777777; stroke-width: 1;"
                  x1="0"
                  x2="-5"
                  y1="0"
                  y2="0"
                />
                <text
                  dominant-baseline="central"
                  style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                  text-anchor="end"
                  transform="translate(-10,0) rotate(0)"
                >
                  1800000
                </text>
              </g>
              <line
                style="stroke: transparent; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="300"
              />
            </g>
            <path
              d="M0,44L125,47L250,38L375,21L500,0"
              fill="none"
              stroke="#e8c1a0"
              stroke-width="2"
            />
            <g>
              <g
                style="pointer-events: none;"
                transform="translate(500, 0)"
              >
                <circle
                  fill="#e8c1a0"
                  r="3"
                  stroke="transparent"
                  stroke-width="0"
                  style="pointer-events: none;"
                />
              </g>
              <g
                style="pointer-events: none;"
                transform="translate(375, 21)"
              >
                <circle
                  fill="#e8c1a0"
                  r="3"
                  stroke="transparent"
                  stroke-width="0"
                  style="pointer-events: none;"
                />
              </g>
              <g
                style="pointer-events: none;"
                transform="translate(250, 38)"
              >
                <circle
                  fill="#e8c1a0"
                  r="3"
                  stroke="transparent"
                  stroke-width="0"
                  style="pointer-events: none;"
                />
              </g>
              <g
                style="pointer-events: none;"
                transform="translate(125, 47)"
              >
                <circle
                  fill="#e8c1a0"
                  r="3"
                  stroke="transparent"
                  stroke-width="0"
                  style="pointer-events: none;"
                />
              </g>
              <g
                style="pointer-events: none;"
                transform="translate(0, 44)"
              >
                <circle
                  fill="#e8c1a0"
                  r="3"
                  stroke="transparent"
                  stroke-width="0"
                  style="pointer-events: none;"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: relative;"
    >
      <svg
        height="300"
        role="img"
        width="500"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="transparent"
          height="300"
          width="500"
        />
        <g
          transform="translate(0,0)"
        >
          <g>
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="0"
              y1="0"
              y2="300"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="125"
              x2="125"
              y1="0"
              y2="300"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="250"
              x2="250"
              y1="0"
              y2="300"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="375"
              x2="375"
              y1="0"
              y2="300"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="500"
              x2="500"
              y1="0"
              y2="300"
            />
          </g>
          <g>
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="300"
              y2="300"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="269"
              y2="269"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="239"
              y2="239"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="208"
              y2="208"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="177"
              y2="177"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="146"
              y2="146"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="116"
              y2="116"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="85"
              y2="85"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="54"
              y2="54"
            />
            <line
              opacity="1"
              stroke="#dddddd"
              stroke-width="1"
              x1="0"
              x2="500"
              y1="24"
              y2="24"
            />
          </g>
          <g
            transform="translate(0,300)"
          >
            <g
              style="opacity: 1;"
              transform="translate(0,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="middle"
                transform="translate(0,10) rotate(0)"
              >
                1960
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(125,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="middle"
                transform="translate(0,10) rotate(0)"
              >
                1965
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(250,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="middle"
                transform="translate(0,10) rotate(0)"
              >
                1970
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(375,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="middle"
                transform="translate(0,10) rotate(0)"
              >
                1975
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(500,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="5"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="middle"
                transform="translate(0,10) rotate(0)"
              >
                1980
              </text>
            </g>
            <line
              style="stroke: transparent; stroke-width: 1;"
              x1="0"
              x2="500"
              y1="0"
              y2="0"
            />
          </g>
          <g
            transform="translate(0,0)"
          >
            <g
              style="opacity: 1;"
              transform="translate(0,300)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                0
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,269)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                200000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,239)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                400000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,208)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                600000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,177)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                800000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,146)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                1000000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,116)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                1200000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,85)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                1400000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,54)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                1600000
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,24)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="-5"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-family: sans-serif; font-size: 11px; fill: #333333;"
                text-anchor="end"
                transform="translate(-10,0) rotate(0)"
              >
                1800000
              </text>
            </g>
            <line
              style="stroke: transparent; stroke-width: 1;"
              x1="0"
              x2="0"
              y1="0"
              y2="300"
            />
          </g>
          <path
            d="M0,44L125,47L250,38L375,21L500,0"
            fill="none"
            stroke="#e8c1a0"
            stroke-width="2"
          />
          <g>
            <g
              style="pointer-events: none;"
              transform="translate(500, 0)"
            >
              <circle
                fill="#e8c1a0"
                r="3"
                stroke="transparent"
                stroke-width="0"
                style="pointer-events: none;"
              />
            </g>
            <g
              style="pointer-events: none;"
              transform="translate(375, 21)"
            >
              <circle
                fill="#e8c1a0"
                r="3"
                stroke="transparent"
                stroke-width="0"
                style="pointer-events: none;"
              />
            </g>
            <g
              style="pointer-events: none;"
              transform="translate(250, 38)"
            >
              <circle
                fill="#e8c1a0"
                r="3"
                stroke="transparent"
                stroke-width="0"
                style="pointer-events: none;"
              />
            </g>
            <g
              style="pointer-events: none;"
              transform="translate(125, 47)"
            >
              <circle
                fill="#e8c1a0"
                r="3"
                stroke="transparent"
                stroke-width="0"
                style="pointer-events: none;"
              />
            </g>
            <g
              style="pointer-events: none;"
              transform="translate(0, 44)"
            >
              <circle
                fill="#e8c1a0"
                r="3"
                stroke="transparent"
                stroke-width="0"
                style="pointer-events: none;"
              />
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Index should render a line chart 1`] = `undefined`;

exports[`Index updates graph based on checkbox 1`] = `Promise {}`;

exports[`Index updates graph based on checkbox 2`] = `Promise {}`;

exports[`Index updates graph based on checkbox 3`] = `Promise {}`;

exports[`Index updates graph based on checkbox 4`] = `Promise {}`;

exports[`Index updates graph based on checkbox 5`] = `Promise {}`;

exports[`Index updates graph based on checkbox 6`] = `Promise {}`;
